src/main/java/org/openstreetmap/atlas/utilities/collections/StreamIterable.java

```diff
@@ -30,6 +30,30 @@ protected StreamIterable(final Iterable<T> source)
         this.source = source;
     }
 
+    /**
+     * Test whether all elements from iterable matches given predicate or not
+     *
+     * @param predicate
+     *            Predicate to test
+     * @return True when given predicate is true for all entities in iterable, else false
+     */
+    public boolean allMatch(final Predicate<T> predicate)
+    {
+        return StreamSupport.stream(this.source.spliterator(), false).allMatch(predicate);
```

Comment:
Any reason not to use parallel stream here?
```
return StreamSupport.stream(this.source.spliterator(), true).allMatch(predicate)
```

Reply:
We need to add parallel method to Iterables' stream, and then we can add parallel method here as well. Some things the client may not want or can have parallelized - so this would give us more flexibility to do that. Defaulting to non-parallel until that functionality is there.